src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/StaticTable.cs

```diff
@@ -2,17 +2,32 @@
 // Licensed under the Apache License, Version 2.0.
 // See THIRD-PARTY-NOTICES.TXT in the project root for license information.
 
+using System.Collections.Generic;
 using System.Text;
 
 namespace System.Net.Http.HPack
 {
     internal static class StaticTable
     {
-        public static int Count => s_staticDecoderTable.Length;
 
-        public static HeaderField Get(int index) => s_staticDecoderTable[index];
+        private static readonly Dictionary<int, int> _statusIndex = new Dictionary<int, int>
+        {
+            [200] = 8,
+            [204] = 9,
+            [206] = 10,
+            [304] = 11,
+            [400] = 12,
+            [404] = 13,
+            [500] = 14,
+        };
```

Comment:
Another case where it'd be helpful to have a comment indicating where this particular list comes from and why it's isolated to just these values.